sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/OpenCoverReportParser.java

```diff
@@ -145,17 +164,54 @@ private void handleBranchPointTag(XmlParserHelper xmlParserHelper) {
         int path = xmlParserHelper.getRequiredIntAttribute("path");
         int visitCount = xmlParserHelper.getRequiredIntAttribute("vc");
 
-        if (fileService.isSupportedAbsolute(filePath)) {
-          coverage.add(new BranchPoint(filePath, line, offset, offsetEnd, path, visitCount));
+        if (coveredFile.isSupported()) {
+          coverage.add(new BranchPoint(coveredFile.supportedPath, line, offset, offsetEnd, path, visitCount));
+
+          LOG.trace("OpenCover parser: add branch hits for file {}, line '{}', offset '{}', visitCount '{}'.",
+            coveredFile, line, offset, visitCount);
         } else {
-          LOG.debug("OpenCover parser: Skipping the fileId '{}', line '{}', vc '{}' because file '{}'" +
+          LOG.debug("OpenCover parser: Skipping branch hits for file {}, line '{}', offset '{}', visitCount '{}' because file" +
               " is not indexed or does not have the supported language.",
-            fileId, line, visitCount, filePath);
+            coveredFile, line, offset, visitCount);
         }
-
       } else {
-        LOG.debug("OpenCover parser: the fileId '{}' key is not contained in files", fileId);
+        LOG.debug("OpenCover parser (handleBranchPointTag): the fileId '{}' key is not contained in files.", fileId);
       }
     }
   }
+
+  /**
+   * Represents a file mentioned in the OpenCover report list of files, which may or may not be supported by the FileService.
+   * @see FileService
+   */
+  private class CoveredFile
+  {
+    // the file ID from the OpenCover report
+    private String uid;
+    // the path from the coverage file
+    private String originalPath;
+    // the path which is supported by the FileService
+    private String supportedPath;
```

Comment:
I thought of this name as well. Indeed, "supported" is a bit too ambiguous.
I also thought about "scannerPath" or "indexedPath", as it needs to be supported by the scanner. 

I'll use "indexedPath" because that's actually what we need for coverage to work: the path needs to be known to the scanner.

Reply:
Sounds good!